backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -166,7 +182,28 @@ func (c *ClientManager) Authenticators() []auth.Authenticator {
 	return c.authenticators
 }
 
+func (c *ClientManager) ControllerClient() ctrlclient.Client {
+	return c.controllerClient
+}
+
 func (c *ClientManager) init() {
+	glog.Info("Initializing controller client...")
+	restConfig, err := util.GetKubernetesConfig()
+	if err != nil {
+		panic(fmt.Sprintf("Failed to initialize the RestConfig: %v", err))
+	}
+	controllerClient, err := ctrlclient.New(
```

Comment:
Optional: It'd be good to add caching since the webhook looks at current pipelines. For example:
https://github.com/mprahl/pipelines/commit/56080be8a0fc44d2f35141ff90ed55bcf8aa8a7d#diff-aba94cb5fe1414981cd5129840e648c04c99096c5370cf2c0dfa44f31babc502R214-R240

That being said, if @rimolive is adding it in his Kubernetes pipeline store PR, then just coordinate with him so it gets reused in the webhook.